### PR TITLE
feat: enable govet in golangci-lint

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: Kush Trivedi <kushthedude@gmail.com>

Refers #1244

### CODE REVIEW

**Comment by wojtek-t** [comment]
@kushthedude - seems there are number of issues that has to be fixed before enabling them

**Comment by kushthedude** [comment]
> @kushthedude - seems there are number of issues that has to be fixed before enabling them

Yes, strange that most of these issues were not reported by govet on local.

**Comment by kushthedude** [comment]
@wojtek-t Commenting out misspell right now as most of the spell errors are in the vendor modules which should not be fixed by us. This should be enabled when we have migrated completely to go modules.

File: .golangci.yml [hunk 1]
```diff
@@ -6,17 +6,28 @@ linters-settings:
...
 linters:
   disable-all: true
   enable:
     - gofmt
     - golint
+    - govet
     - errcheck
     - deadcode
-    # TODO(oxddr): enable more checkers
+    # Todo(kushthedude) fix after vendor is migrated to go modules
```
Comment (wojtek-t) [inline .golangci.yml:27]: nit: TODO (not Todo)

File: .golangci.yml [hunk 2]
```diff
@@ -6,17 +6,28 @@ linters-settings:
...
     # - misspell
 
+
```
Comment (wojtek-t) [inline .golangci.yml:30]: nit: remove this empty line

File: .golangci.yml [hunk 3]
```diff
@@ -6,15 +6,28 @@ linters-settings:
...
+  govet:
+    check-shadowing: false
+    settings:
+      printf:
+        funcs:
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Infof
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Warnf
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Errorf
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Fatalf
+    disable:
```
Comment (wojtek-t) [inline .golangci.yml:18]: Can you add a TODO to get rid of those disabled ones?

Comment (kushthedude) [reply .golangci.yml]: Done!

**Comment by wojtek-t** [comment]
Something doesn't work now - please fix

**Comment by kushthedude** [comment]
> Something doesn't work now - please fix

Was an error in linting the configuration file, fixed now!

**Pull request merged**